### PR TITLE
MIAUW VON SAYCODE 3: RISE OF THE SPANS

### LINKED ISSUE #5833: HTML tags showing on the station bounced radio when using ' or " and trying to ;speak on the radio.
I had a station bounced radio in my backpack with speaker and microphone on.
I took my mic off.

Tested with multiple punctuation, ' and " are the only ones that do it that I can see.

The specific message typed was:

;Tes'ting
;Tes"ting

Outputted as:

Doesn't happen if you don't use ; which station bounced radios don't need anyway, but eh.

### PR DESCRIPTION
**Pull request by theOperand**
i have begun to de-shittify how spans are handled in saycode.
completely untested right now, but it compiles.
also de-shittifies the say_quotes() and adds three new variables: verb_say, verb_ask and verb_yell.

should fix #5833

this also makes it possible to have spans manipulated in NTSL, but i dont into NTSL.

also shout-out to gia for saying that we needed spans way before anybody else did.

### CODE REVIEW

**Comment by RemieRichards** [comment]
>verb_ask

Cmon miauw the word is Query.

**Comment by Miauw62** [comment]
this makes more sense :V

**Comment by GunHog** [comment]
Why not add the Goofball spans to this? It seems you laid the proper groundwork to make that somewhat simpler.

**Comment by Miauw62** [comment]
goofball added spans?

**Comment by Iamgoofball** [comment]
Make sure the comic sans span class works with this system.

**Comment by GunHog** [comment]
It is the 'sans' span class, applied here: https://github.com/tgstation/-tg-station/commit/39ede45b31725a9426bbc780a77209a07f346301#diff-d71684bfd3d949193b7ed53220fc8473R148

For silicons, it is the 'robot' span class, which it seems you already found and added. The other stuff goof added are word replacements.

**Comment by Miauw62** [comment]
DONE

File: code/game/say.dm [hunk 1]
```diff
@@ -106,27 +112,49 @@ var/list/freqtospan = list(
 		return returntext
 	return "[copytext("[freq]", 1, 4)].[copytext("[freq]", 4, 5)]"
 
+/proc/attach_spans(input, list/spans = list())
+	return "[message_spans_start(spans)][input][message_spans_end(spans.len)]"
+
+/proc/message_spans_start(list/spans = list())
+	var/output = ""
+	for(var/S in spans)
```
Comment (Giacom) [inline code/game/say.dm]: You don't need a list of spans. You just need a list of classes to use as you can combine them like so:

```HTML
<span class="notice warning">Blah</span>
```

Comment (Miauw62) [inline code/game/say.dm]: Oooh, thanks. I don't into HTML/CSS.

**Comment by Miauw62** [comment]
sigh, this is not working because the spans list is turning into null fnr.
```
runtime error: Cannot read null.len
proc name: message spans start (/proc/message_spans_start)
  source file: say.dm,120
  usr: Meghan Anderson (/mob/living/carbon/human)
  src: null
  call stack:
message spans start(null)
attach spans("Test", null)
Meghan Anderson (/mob/living/carbon/human): say quote("Test", null)
Meghan Anderson (/mob/living/carbon/human): say quote("Test", null)
Meghan Anderson (/mob/living/carbon/human): say quote("Test", null)
Meghan Anderson (/mob/living/carbon/human): lang treat(Meghan Anderson (/mob/living/carbon/human), 1, "Test", null)
Meghan Anderson (/mob/living/carbon/human): compose message(Meghan Anderson (/mob/living/carbon/human), 1, "Test", null, null)
Meghan Anderson (/mob/living/carbon/human): send speech("Test", 7, Meghan Anderson (/mob/living/carbon/human), null, null, /list (/list))
Meghan Anderson (/mob/living/carbon/human): say("Test", null)
Meghan Anderson (/mob/living/carbon/human): Say("Test")
```
FYI i added ```world << spans.len``` to say_quote for debugging, this runtime resulted. As you can see, spans is becoming null when calling compose_message() in send_speech()

**Comment by Miauw62** [comment]
fuck i just had one argument too much somewhere.
i swear misaligned args are like 90% of my troubles with saycode, i wish BYOND didnt let you do this sort of silly stuff.

**Comment by Miauw62** [comment]
Alright, this works now, with a complementary bugfix on the side.

**Comment by Miauw62** [comment]
Somebody review this and then merge it or something, vOv

Also somebody else will need to make NTSL access for this.

**Comment by GunHog** [comment]
Oh, do not worry, dear Meow. I will handle that :)

**Comment by optimumtact** [comment]
@Cheridan @Giacom @RemieRichards @phil235 @Razharas 

merge please, I'll send you lots of love and somethign special

**Comment by RemieRichards** [comment]
What I know of saycode, this is fine, Plus Miauw himself says its fine so there we go.

**Comment by optimumtact** [comment]
Based :sun_with_face:

**Comment by GunHog** [comment]
BASED INDEED! Time to add this to NTSL :D

**Pull request merged**